cmd/cluster-agent/subcommands/start/command.go

```diff
@@ -474,15 +475,19 @@ func start(log log.Component,
 		}
 
 		webhooks, err := admissionpkg.StartControllers(admissionCtx, wmeta, pa)
-		if err != nil {
+		// Ignore the error if it's related to the missing validatingwebhookconfigurations RBACs
+		if err != nil && !strings.HasPrefix(err.Error(), "couldn't sync informer admissionregistration.k8s.io/v1/validatingwebhookconfigurations") {
```

Comment:
If we need to make the distinction between several types of errors, I think it would be more robust to use several error types rather than relying on free-form text.

We would need to create a dedicated error type here: https://github.com/DataDog/datadog-agent/blob/2c2eb2293cccd33100d7d930a59c136319942915/pkg/util/kubernetes/apiserver/util.go#L45

Something like:
```go
type SyncInformersError {
	Name    string
	Elapsed time.Duration
}

func (e *SyncInformersError) Error() string {
	return fmt.Sprintf("couldn't sync informer %s in %v", e.name, e.elapsed)
}
```

Then, [where the error is emitted](https://github.com/DataDog/datadog-agent/blob/2c2eb2293cccd33100d7d930a59c136319942915/pkg/util/kubernetes/apiserver/util.go#L45):
```diff
-				return fmt.Errorf("couldn't sync informer %s in %v", name, end.Sub(start))
+				return SyncInformersError{Name: name, Elapsed: end.Sub(start)}
```

With such a dedicated error type, we could assert it with a more robust manner:

```suggestion
		if err != nil && errors.As(err, &apiserver.SyncInformersError{}) {
```

Reply:
Thanks! I've improved things in this commit: https://github.com/DataDog/datadog-agent/pull/28512/commits/1867d8fe172a8186942fb2287f1bd4c9eb96a2a5 🙏